testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportImproperlyOverriddenTestMethods.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testng.services;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.junit.platform.commons.annotation.Testable;
+import org.testng.IClassListener;
+import org.testng.ITestClass;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.testng.services.Listeners.reportListenerFailure;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.util.stream.Collectors.joining;
+import static org.junit.platform.commons.util.AnnotationUtils.isAnnotated;
+
+/**
+ * This checker finds issues in JUnit 5 tests where the test method is overridden in a subclass but
+ * not annotated with {@link org.junit.jupiter.api.Test} or any other {@link Testable}. In addition
+ * to presence of the annotation, this check requires that the overridden method is annotated with the
+ * same annotation with the same parameters, e.g. a {@code @RepeatedTest(5)} method's override needs
+ * to be also annotated with {@link org.junit.jupiter.api.RepeatedTest} with the same number of repeats (5).
```

Comment:
Is this too strict? I can imagine cases where e.g. a simple test gets overridden as a repeated test, or something like this.

Reply:
We may want to override to _change_ some test attributes.